### PR TITLE
Remove unnecessary @Nullable annotations

### LINKED ISSUE #1145: Request headers shouldn't be `@Nullable`
First of all, thanks for your huge work on this project :heart: 

I have a question about `@Nullable` in the `headers` field in the `BaseRequestConfiguration` class.

In what situations could this field be nullable? This requires API consumers to implement null checks, as it raises the following warnings:

Screenshot from 2024-03-31 17-58-20

This should be an empty object, which cannot be `null`.

### PR DESCRIPTION
**Pull request by piotrooo**
Closes #1145

### CODE REVIEW

**Comment by piotrooo** [comment]
@microsoft-github-policy-service agree

File: components/abstractions/src/main/java/com/microsoft/kiota/BaseRequestConfiguration.java [hunk 1]
```diff
@@ -13,9 +14,13 @@ public BaseRequestConfiguration() {
         // default empty constructor
     }
 
-    /** Request headers */
-    @Nullable public RequestHeaders headers = new RequestHeaders();
+    /**
+     * Request headers
+     */
+    public RequestHeaders headers = new RequestHeaders();
```
Comment (andrueastman) [inline components/abstractions/src/main/java/com/microsoft/kiota/BaseRequestConfiguration.java:20]: I believe the annotation should change to `@NonNull`

**Review by andrueastman — COMMENTED** [review]
Any chance you can also add a entry to the changelog file at the root of the repositoy as well?

**Review by andrueastman — APPROVED** [review]
Thanks for the contribution @piotrooo

**Pull request merged**